autopilot/accounts.go

```diff
@@ -119,144 +107,131 @@ func (a *accounts) refillWorkerAccounts(w Worker) {
 	ctx, span := tracing.Tracer.Start(context.Background(), "refillWorkerAccounts")
 	defer span.End()
 
+	// fetch config
+	cfg := a.ap.Config()
+
+	// fetch worker id
 	workerID, err := w.ID(ctx)
 	if err != nil {
 		span.RecordError(err)
 		span.SetStatus(codes.Error, "failed to fetch worker id")
-		a.logger.Errorw(fmt.Sprintf("failed to fetch worker id for refill: %v", err))
+		a.l.Errorw(fmt.Sprintf("failed to fetch worker id for refill: %v", err))
 		return
 	}
 
-	for hk, contracts := range a.contractsToRefill() {
-		// skip if refill is in progress
-		if !a.markRefillInProgress(workerID, hk) {
-			continue
-		}
-
-		// launch a goroutine to refill the account
-		go func(hk types.PublicKey, contracts []api.ContractMetadata) error {
-			defer a.markRefillDone(workerID, hk)
-
-			// decide whether one of the potentially multiple contracts is in
-			// the contract set, we only want to log errors if that is the case
-			// because we expect refills to fail often for contracts that are
-			// not in the set
-			var hasContractSetContract bool
-			for _, contract := range contracts {
-				if a.isContractSetContract(contract.ID) {
-					hasContractSetContract = true
-					break
-				}
-			}
+	// fetch all contracts
+	contracts, err := a.c.Contracts(ctx)
+	if err != nil {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, "failed to fetch contracts")
+		a.l.Errorw(fmt.Sprintf("failed to fetch contracts for refill: %v", err))
+		return
+	}
 
-			// make sure we don't hang
-			ctx, cancel := context.WithTimeout(ctx, time.Minute)
-			defer cancel()
+	// prepare no-op logger
+	noop := zap.NewNop().Sugar()
+	logger := noop
 
-			// add tracing
-			ctx, span := tracing.Tracer.Start(ctx, "refillAccount")
-			defer span.End()
-			defer func() {
-				if err != nil {
-					span.RecordError(err)
-					span.SetStatus(codes.Error, "failed to refill account")
+	// refill accounts in separate goroutines
+	for _, c := range contracts {
+		// skip contracts that are up for renewal
```

Comment:
I don't think this is a good idea. There is no harm in trying to refill with a contract that is up for renewal or refreshing. This is going to achieve that we will have a potentially very long gap where we don't refill any an accounts anymore since we entered the renew window.

Ideally that fails to renew because we don't need it won't end up in the contract set and therefore not cause any log spam.